RecoEgamma/EgammaTools/src/LowPtConversion.cc

```diff
@@ -0,0 +1,212 @@
+    if (conv.tracksPin().size() >= 2 && ilead_ > -1 && itrail_ > -1) {
+      math::XYZVectorF lead_Pin = conv.tracksPin().at(ilead_);
+      math::XYZVectorF trail_Pin = conv.tracksPin().at(itrail_);
+      mass_from_Pin_ = mee(lead_Pin.x(), lead_Pin.y(), lead_Pin.z(), trail_Pin.x(), trail_Pin.y(), trail_Pin.z());
+      // Opening angle
+      delta_cot_from_Pin_ = 1. / tan(trail_Pin.theta()) - 1. / tan(lead_Pin.theta());
+    }
+
+    // Invariant mass before fit to common vertex
+    if (conv.tracks().size() >= 2 && ilead_ > -1 && itrail_ > -1) {
+      auto lead_before_vtx_fit = conv.tracks().at(ilead_)->momentum();
+      auto trail_before_vtx_fit = conv.tracks().at(itrail_)->momentum();
+      mass_before_fit_ = mee(lead_before_vtx_fit.x(),
+                             lead_before_vtx_fit.y(),
+                             lead_before_vtx_fit.z(),
+                             trail_before_vtx_fit.x(),
+                             trail_before_vtx_fit.y(),
+                             trail_before_vtx_fit.z());
+    }
+
+    // Invariant mass after the fit to common vertex
+    if (conv.conversionVertex().refittedTracks().size() >= 2 && ilead_ > -1 && itrail_ > -1) {
+      auto const& lead_after_vtx_fit = conv.conversionVertex().refittedTracks().at(ilead_);
+      auto const& trail_after_vtx_fit = conv.conversionVertex().refittedTracks().at(itrail_);
+      mass_after_fit_ = mee(lead_after_vtx_fit.px(),
+                            lead_after_vtx_fit.py(),
+                            lead_after_vtx_fit.pz(),
+                            trail_after_vtx_fit.px(),
+                            trail_after_vtx_fit.py(),
+                            trail_after_vtx_fit.pz());
+      // Difference in expeted hits
+      delta_expected_nhits_inner_ =
+          lead_after_vtx_fit.hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS) -
+          trail_after_vtx_fit.hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS);
+    }
+
+    // Hits prior to vertex
+    if (ilead_ > -1 && itrail_ > -1) {
+      auto const& nHits = conv.nHitsBeforeVtx();
+      bool enoughTracks = nHits.size() > 1;
+      lead_nhits_before_vtx_ = enoughTracks ? nHits.at(ilead_) : 0;
+      trail_nhits_before_vtx_ = enoughTracks ? nHits.at(itrail_) : 0;
+      max_nhits_before_vtx_ = enoughTracks ? std::max(nHits[0], nHits[1]) : 0;
+      sum_nhits_before_vtx_ = enoughTracks ? nHits[0] + nHits[1] : 0;
+    }
+
+    // Attempt to match conversion track to electron
+    for (uint itrk = 0; itrk < conv.tracks().size(); ++itrk) {
+      const edm::RefToBase<reco::Track> trk = conv.tracks()[itrk];
+      if (trk.isNull()) {
+        continue;
+      }
+      reco::GsfTrackRef ref = ele.core()->gsfTrack();
+      reco::GsfTrackRef gsf = ele.gsfTrack();
+      if (gsf.isNull()) {
+        continue;
+      }
+      if (ref.id() == trk.id() && ref.key() == trk.key()) {
+        matched_ = true;
+        if ((int)itrk == ilead_) {
```

Comment:
```suggestion
        if (static_cast<int>(itrk) == ilead_) {
```
4.15 in http://cms-sw.github.io/cms_coding_rules.html#4--technical-coding-rules-1

Reply:
Done.